lib/handlers/injector-sor.ts

```diff
@@ -108,9 +108,20 @@ export abstract class InjectorSOR<Router, QueryParams> extends Injector<
     const dependenciesByChainArray = await Promise.all(
       _.map(SUPPORTED_CHAINS, async (chainId: ChainId) => {
         const chainName = ID_TO_NETWORK_NAME(chainId)
-        // updated chainNames to match infura strings
+        const providerName = process.env.PROVIDER_NAME
         const projectId = process.env.PROJECT_ID
-        const url = `https://${chainName}.infura.io/v3/${projectId}`
+        let url = ''
+        switch (providerName) {
+          case 'infura':
+            url = `https://${chainName}.infura.io/v3/${projectId}`
+            break
+          case 'alchemy':
+          // todo: add logic to generate alchemy rpc urls
+          case 'moralis':
+          // todo: todo: add logic to generate moralis rpc urls
```

Comment:
Could you fill in alchemy just so its ready if we need to switch out?

Think we can remove moralis too for now